frontend/src/modules/editor/components/Editor.js

```diff
@@ -100,6 +101,27 @@ export class EditorBase extends React.Component<InternalProps> {
         );
     }
 
+    getFailedChecksOfType = (type: string) => {
+        const { editor, pluralForm, selectedEntity } = this.props;
+
+        if (!selectedEntity) {
+            return [];
+        }
+
+        if (editor[type].length) {
+            return editor[type];
+        }
+
+        const plural = pluralForm === -1 ? 0 : pluralForm;
+        const translation = selectedEntity.translation[plural];
+
+        if (!translation || (!translation.approved && !translation.fuzzy)) {
```

Comment:
We only show checks of the active translation if it's approved or fuzzy (because than its status is also colored as error/warning in the string list).

But now that I think about it, maybe we should just show the failed checks by default anyways? Otherwise the user might think it's OK to approve the translation, and once she approves it, she'll see the popup (which is not an optimal UX).

What do you think?

Reply:
That sounds reasonable indeed, though maybe we should refrain from changing the behavior from Translate.Current for now?

Reply:
Good point! I'll add a comment.